src/sqlancer/DatabaseProvider.java

```diff
@@ -32,6 +32,18 @@
      */
     Reproducer<G> generateAndTestDatabase(G globalState) throws Exception;
 
+    /**
+     * The experimental feature: QPG.
+     *
+     * @param globalState
+     *            the state created and is valid for this method call.
+     *
+     * @throws Exception
+     *             if testing fails.
+     *
+     */
+    void qpg(G globalState) throws Exception;
```

Comment:
I think `qpg` is quite cryptic for people not already familiar with the technique. Could we use another method name, like `generateAndTestDatabaseWithQueryGuidance`?